controllers/pipelines/reconciler.go

```diff
@@ -31,6 +32,21 @@ func desiredProvider(resource pipelinesv1.HasProvider, config config.Configurati
 	return config.DefaultProvider
 }
 
+func loadProvider(ctx context.Context, reader client.Reader, namespace string, desiredProvider string) (pipelinesv1.Provider, error) {
```

Comment:
as this is used in several controllers, it should live in a central place. You could make it a method on the Base Reconciler (might mean you don't need to pass in `client.Reader`)

Reply:
updated in [dab5cdb](https://github.com/sky-uk/kfp-operator/pull/354/commits/dab5cdb80ee005d7ff3a19d1f4f115a953485898)